R/give_opinions.R

```diff
@@ -1,127 +1,169 @@
-  }else{
-    checkurls <- check_urls(descr$get("BugReports"))
-    if(checkurls != ""){
-      bugreports_fixme <- checkurls
-    }else{
-      bugreports_fixme <- NULL
-    }
+  fixmes
+}
 
+# fixmes_as_tibble_or_message ---------------------------------------------------
+fixmes_as_tibble_or_message <- function(fixmes, where, message_id = NULL) {
+
+  if (length(fixmes)) {
+
+    tibble::tibble(where = where, fixme = fixmes)
+
+  } else if (! is.null(message_id)) {
+
+    message(get_message(message_id))
   }
+}
+
+# try_to_give_opinions_readme --------------------------------------------------
+try_to_give_opinions_readme <- function(description_file) {
 
-  fixmes <- c(authors_fixme, url_fixme, bugreports_fixme)
-  fixmes <- fixmes[!is.null(fixmes)]
+  readme_path <- guess_readme(dirname(description_file))$readme_path
 
-  if(length(fixmes) >0){
+  if (is.null(readme_path)) {
 
-    tibble::tibble(where = "DESCRIPTION",
-                   fixme = fixmes)
-  }else{
-    NULL
+    return(NULL)
   }
 
+  desc_info <- codemeta_description(description_file)
 
+  give_opinions_readme(readme_path, pkg_name = desc_info$identifier)
 }
 
-give_opinions_readme <- function(readme_path,
-                                 pkg_name){
+# give_opinions_readme ---------------------------------------------------------
+give_opinions_readme <- function(readme_path, pkg_name) {
+
+  # start with an empty vector of "fixme" messages
+  fixmes <- character()
 
-  # look for badges
-  badges <- extract_badges(readme_path)
   # status
-  if(is.null(guess_devStatus(readme_path))){
-    status_fixme <- "Add a status badge cf e.g repostatus.org"
-  }else{
-    status_fixme <- NULL
+  if (is.null(guess_devStatus(readme_path))) {
+
+    fixmes <- c(fixmes, get_message("hint_add_status_badge"))
   }
 
   # provider
   provider <- guess_provider(pkg_name)
-  if(!is.null(provider)){
-    provider_badge <- whether_provider_badge(badges,
-                                             provider$name)
-
-    if(!provider_badge){
-      provider_fixme <- paste0("There is a package called ",
-                               pkg_name, " on ",
-                               provider$name, ", add a badge to show it is yours or rename your package.")
-    }else{
-      provider_fixme <- NULL
-    }
-  }else{
-    provider_fixme <- NULL
-  }
 
-  fixmes <- c(status_fixme, provider_fixme)
-  fixmes <- fixmes[!is.null(fixmes)]
-  if(length(fixmes) >0){
+  if (has_provider_but_no_batch(provider, readme_path)) {
 
-    tibble::tibble(where = "README",
-                   fixme = fixmes)
-  }else{
-    message("codemetar has the highest opinion of this R package :-)")
-    NULL
+    fixmes <- c(fixmes, get_message(
+      "hint_package_exists", pkg_name, provider$name
+    ))
   }
 
+  fixmes_as_tibble_or_message(fixmes, "README", "hint_highest_opinion")
+}
+
+# has_provider_but_no_batch ----------------------------------------------------
```

Comment:
I don't understand the name of this function?

Reply:
Oooh I think you meant `has_provider_but_no_badge`!

Reply:
Indeed... badge, batch, who cares? 
I will rename!

Reply:
Depending on the accent there is a difference 😹